cql3/restrictions/statement_restrictions.cc

```diff
@@ -807,7 +807,7 @@ bool limits(const binary_operator& opr, const column_value_eval_bag& bag) {
     } else if (columns.size() == 1) {
         auto lhs = get_value(columns[0], bag);
         if (!lhs) {
-            lhs = bytes(); // Compatible with old code, which feeds null to type comparators.
```

Comment:
What is this old code that empty value was ensuring compatibility with?

Reply:
I presume it was "the code before Dejan's overhaul", which passes NULLs as empty buffers, being exactly the source of this bug.

Reply:
That's correct.